projects/004-exercise-tracker/jimmyhu/src/api/users/usersapi.ts

```diff
@@ -0,0 +1,125 @@
+const exps: any = require('express')
+
+const router = exps.Router()
+
+
+// global variables for store the information
+let users: user[] = []
+let exercises: exercise[] = []
+let invalidID:  number[] = []
+
+
+// user interface, the Id will be a random number for now
+interface user {
+    'username' : string,
+    '_id': number
+}
+
+// exercise extend the user interface due the structure require the same information anyway
+interface exercise extends user {
+    "description": string,
+    "duration": number,
+    "date": Date|string,
+}
+
+// function for check if the date is in the interval
+function checkdate (Indate:string|Date, start: string, end: string): boolean{
+    if (new Date(Indate) > new Date(start) && new Date(Indate) < new Date(end)){
+        return true
+    } else if (start == undefined || end == undefined){
+        return true
+    }
+    return false
+}
+
+
+// base post request, the id is a random number auto-generated, for avoid double id there are a check on a list
+router.post('/', (req: any, res:any): void => {
+    let username: string = req.body.name
+    let randID: number = Math.floor(Math.random() * 1000)
+    while (invalidID.includes(randID)){
+        let randID: number = Math.floor(Math.random() * 1000)
+    }
+    let newUser: user = {
+        'username' : username,
+        '_id': randID
+    }
+    invalidID.push(randID)
+    users.push(newUser)
+    res.json(newUser)
+})
+
+router.get('/', (req: any, res:any): void => {
+    res.json(users)
+})
+
+router.post('/:_id/exercises', (req: any, res:any): void => {
+    // check if the id exist in the stored list
+    if(!invalidID.includes(parseInt(req.params._id))){
+        res.send('Invalid ID')
+        return
+    }
+    if(new Date(req.body.date).toUTCString() == null || new Date(req.body.date).toUTCString() == 'Invalid Date'){
+        res.send('Invalid Date')
+        return
+    }
+    // check if the required information is different from undefinied
+    if(req.body.description == undefined || req.body.duration == undefined){
+        res.send("Missing Information")
+        return
+    }
+    // find the id that match in the list of registered user
+    let username:string = ''
```

Comment:
Ci potrebbe essere un modo più elegante e leggibile per trovare l'utente. Ti viene nulla in mente?